e2e/tcproute_test.go

```diff
@@ -680,7 +681,7 @@ func TestMeshWithMultipleTCPRoutes(t *testing.T) {
 	processGraphAndExpectActions(t, graphBuilder, expectedActions)
 
 	bs, err := theCloud.BackendServices().Get(ctx, removedBS.Key)
-	if err == nil {
+	if !strings.Contains(err.Error(), "not found") {
```

Comment:
we should make this a helper func.

`func errStrNotFound(err error) bool { ... }`

Reply:
Can you use google api to check the error?
```
func IsNotFoundError(err error) bool {
	var gerr *googleapi.Error
	if errors.As(err, &gerr) {
		return gerr.Code == http.StatusNotFound
	}
        return false
}
```

Reply:
Done.